go-controller/pkg/ovn/pods.go

```diff
@@ -321,38 +329,62 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 			}
 		}
 
-		addresses = podMac.String() + " " + util.JoinIPNetIPs(podIfAddrs, " ")
-		args = []string{
-			"--may-exist", "lsp-add", logicalSwitch, portName,
-			"--", "lsp-set-addresses", portName, addresses,
-		}
 	}
-	args = append(args, "--", "set", "logical_switch_port", portName, "external-ids:namespace="+pod.Namespace, "external-ids:pod=true")
-	args = append(args, "--", "lsp-set-port-security", portName, addresses)
 
-	out, stderr, err = util.RunOVNNbctl(args...)
+	// set addresses on the port
+	addresses = make([]string, len(podIfAddrs)+1)
+	addresses[0] = podMac.String()
+	for idx, podIfAddr := range podIfAddrs {
+		addresses[idx+1] = podIfAddr.IP.String()
+	}
+
+	lspSetAddrCmd, err := oc.ovnNBClient.LSPSetAddress(portName, addresses...)
 	if err != nil {
-		return fmt.Errorf("error while creating logical port %s stdout: %q, stderr: %q (%v)",
-			portName, out, stderr, err)
+		return fmt.Errorf("unable to create LSPSetAddress command for port: %s", portName)
+	}
+	cmds = append(cmds, lspSetAddrCmd)
+
+	// add external ids
+	extIds := map[string]string{"namespace": pod.Namespace, "pod": "true"}
+
+	lspSetExtIdsCmd, err := oc.ovnNBClient.LSPSetExternalIds(portName, extIds)
+	if err != nil {
+		return fmt.Errorf("unable to create LSPSetAddress command for port: %s", portName)
+	}
+
+	cmds = append(cmds, lspSetExtIdsCmd)
+
+	// add port security addresses
+	lspPortSecurityCmd, err := oc.ovnNBClient.LSPSetPortSecurity(portName, addresses...)
+	if err != nil {
+		return fmt.Errorf("unable to create LSPSetPortSecurity command for port: %s", portName)
+	}
+
+	cmds = append(cmds, lspPortSecurityCmd)
+
+	// execute all the commands together.
+	err = oc.ovnNBClient.Execute(cmds...)
+	if err != nil {
+		return fmt.Errorf("error while creating logical port %s error: %v",
+			portName, err)
 	}
 
 	// UUID must be retrieved separately from the lsp-add transaction since
 	// (as of OVN 2.12) a bogus UUID is returned if they are part of the same
 	// transaction.
 	// FIXME: move to the lsp-add transaction once https://bugzilla.redhat.com/show_bug.cgi?id=1806788
 	// is resolved.
-	var uuid string
-	uuid, _, err = util.RunOVNNbctl("get", "logical_switch_port", portName, "_uuid")
-	if err != nil {
-		return fmt.Errorf("error while getting UUID for logical port %s "+
-			"stdout: %q, stderr: %q (%v)", portName, uuid, stderr, err)
+	lsp, err = oc.ovnNBClient.LSPGet(portName)
```

Comment:
This may or may not be true anymore... we should investigate whether the bindings do the right thing and wait for the transaction result, and thus have the new UUID. But we can leave that for later.

Reply:
It will fail the Execute() if the transaction fails. So the caller does get an inline failure. I removed the check for UUID altogether.